src/pages/Group/GroupFeed.tsx

```diff
@@ -0,0 +1,184 @@
+import { Group } from "../../data/models/Group";
+import Spinner from "../../components/Spinner";
+import { useApi } from "../../contexts/ApiProvider";
+import { useInfiniteQuery, useQuery } from "react-query";
+import { QueryKeys } from "../../constants";
+import { Happiness, HappinessPagination } from "../../data/models/Happiness";
+import FeedCard from "./FeedCard";
+import HappinessViewerModal from "../../components/modals/HappinessViewerModal";
+import React, { useEffect, useMemo, useState } from "react";
+import { useInView } from "react-intersection-observer";
+import { dateFromStr, formatDate, modifyDateDay } from "../../utils";
+
+// Returns "today" or "yesterday" if the given date is today or yesterday
+// Otherwise returns the default string
+export const dateOrTodayYesterday = (date: string, otherwise: string) => {
+  if (date === formatDate(new Date())) return "Today";
+  else if (date === formatDate(modifyDateDay(new Date(), -1)))
+    return "Yesterday";
+  else return otherwise;
+};
+
+// The group feed first shows the user's unread entries in the group (which are marked as read as they are scrolled past)
+// followed by an infinite scroll of all the group's entries
+export default function GroupFeed({ groupData }: { groupData: Group }) {
+  const { api } = useApi();
+  const [selectedEntry, setSelectedEntry] = useState<Happiness>();
+
+  // for infinite scroll
+  const [bottomRef, bottomInView] = useInView();
+
+  // query for fetching unread entries
+  const unreadQuery = useQuery<Happiness[]>({
+    queryKey: [QueryKeys.FETCH_GROUP_HAPPINESS_UNREAD, groupData.id],
+    queryFn: () =>
+      api
+        .get<Happiness[]>(`/group/${groupData.id}/happiness/unread`)
+        .then((res) => res.data),
+    cacheTime: 0, // disable cache since this causes stale reads to be shown on initial load
+  });
+
+  // infinite query for fetching regular entries
+  const feedQuery = useInfiniteQuery<HappinessPagination>({
+    queryKey: [QueryKeys.FETCH_GROUP_HAPPINESS, groupData.id],
+    queryFn: ({ pageParam = 1 }) =>
+      api
+        .get<Happiness[]>(`/group/${groupData.id}/happiness/count`, {
+          page: pageParam,
+        })
+        .then((res): HappinessPagination => {
+          return {
+            data: res.data,
+            page: pageParam,
+          };
+        }),
+    getNextPageParam: (lastPage: HappinessPagination) => {
+      // no more pages left if the last page is empty
+      if (lastPage.data.length === 0) return false;
+      return lastPage.page + 1; // increment page number to fetch
+    },
+  });
+
+  // flatten React query pagination object into single list
+  const allFeedEntries = useMemo(
+    () =>
+      feedQuery.data?.pages.reduce(
+        (acc: Happiness[], page) => [...acc, ...page.data],
+        [],
+      ),
+    [feedQuery.data],
+  );
+
+  const entryToJsx = (
```

Comment:
I like this idea of making this function to resolve the header issue. Do you think you could make the name a bit more descriptive so it's more readable as to what this section of your code does?